backend/src/layer/data/flickr_api/json_parser.rs

```diff
@@ -182,16 +183,49 @@ mod test {
                 LicenceHistory {
                     date_change: 1_295_918_034,
                     new_license: String::from("Attribution License"),
+                    old_license: String::from("Public Domain Mark"),
                 },
                 LicenceHistory {
                     date_change: 1_598_990_519,
                     new_license: String::from("Public Domain Mark"),
+                    old_license: String::from("Attribution License"),
                 },
             ],
         };
         assert!(JsonParser::check_license(valid_licenses).is_ok());
     }
 
+    #[test]
+    fn test_valid_check_old_license() {
+        let valid_licenses = JsonRootLicense {
+            license_history: vec![LicenceHistory {
+                date_change: 1_295_918_034,
+                new_license: String::new(),
+                old_license: String::from("Public Domain Mark"),
+            }],
+        };
+        assert!(JsonParser::check_license(valid_licenses).is_ok());
+    }
+
+    #[test]
+    fn test_invalid_check_old_license() {
+        let valid_licenses = JsonRootLicense {
+            license_history: vec![
+                LicenceHistory {
+                    date_change: 1_295_918_034,
+                    new_license: String::from("United States Government Work"),
+                    old_license: String::from("Public Domain Dedication (CC0)"),
+                },
+                LicenceHistory {
+                    date_change: 1_598_990_519,
+                    new_license: String::from("All Rights Reserved"),
+                    old_license: String::from("United States Government Work"),
+                },
+            ],
+        };
+        assert!(JsonParser::check_license(valid_licenses).is_err());
+    }
```

Comment:
Short info:
As you can see, only the last history will be checked. 
So only the last old_license and the last new_license is used for validation